skdate/src/date.sk

```diff
@@ -0,0 +1,757 @@
+module SKDate;
+
+@cpp_extern("SKIP_localetimezone")
+native fun localeTimeZone(year: UInt32, month: UInt32, day: UInt32): Int32;
+
+@cpp_extern("SKIP_currenttimemillis")
+native fun natCurrentTimeMillis(): Int;
+
+@cpp_extern("SKIP_JS_currenttimemillis")
+native fun jsCurrentTimeMillis(): Float;
+
+fun currentTimeMillis(): Int {
+  if (SkargoVersion.kWasm32) {
+    (jsCurrentTimeMillis() * 1000.0).toInt()
+  } else {
+    natCurrentTimeMillis()
+  }
+}
+
+class File(value: Calendar) extends SKStore.File
+
+const date_maxDaysInMonth: Array<Int> = Array[
```

Comment:
Everywhere this is used, it's not as a "maximum" but to get the number of days in a month as in
```
        date_maxDaysInMonth[month] +
        if (month == 1 && this.leapYear(year)) {
          1
        } else {
          0
        }
      );
```
Could I suggest instead to factor that all out as something like the following?
```
private const daysInNonLeapYearMonth : Array<Int> = [...];
private static fun daysInMonth(month: Int, year: Int): Int {
 days = daysInMonth[month]
 if ( month == 1  && isLeapYear(year) ) { days + 1 } else { days } 
 }
}
```